packages/dd-trace/test/appsec/passport.spec.js

```diff
@@ -0,0 +1,239 @@
+    username: 'Test User'
+  }
+
+  let passportModule, log, events, setUser
+  beforeEach(() => {
+    log = {
+      warn: sinon.stub()
+    }
+
+    events = {
+      trackEvent: sinon.stub()
+    }
+
+    setUser = {
+      setUserTags: sinon.stub()
+    }
+
+    passportModule = proxyquire('../../src/appsec/passport', {
+      '../log': log,
+      './sdk/track_event': events,
+      './sdk/set_user': setUser
+    })
+  })
+
+  describe('passportTrackEvent', () => {
+    it('should call log when credentials is undefined', () => {
+      passportModule.passportTrackEvent(undefined, undefined, undefined, 'safe')
+
+      expect(log.warn).to.have.been.calledOnceWithExactly('No username found in authentication instrumentation')
+    })
+
+    it('should call log when type is not known', () => {
+      const credentials = { type: 'unknown', username: 'test' }
+
+      passportModule.passportTrackEvent(credentials, undefined, undefined, 'safe')
+
+      expect(log.warn).to.have.been.calledOnceWithExactly('No username found in authentication instrumentation')
+    })
+
+    it('should call log when type is known but username not present', () => {
+      const credentials = { type: 'unknown' }
+
+      passportModule.passportTrackEvent(credentials, undefined, undefined, 'safe')
+
+      expect(log.warn).to.have.been.calledOnceWithExactly('No username found in authentication instrumentation')
+    })
+
+    it('should report login failure when passportUser is not present', () => {
+      passportModule.passportTrackEvent(loginLocal, undefined, undefined, 'safe')
+
+      expect(setUser.setUserTags).not.to.have.been.called
+      expect(events.trackEvent).to.have.been.calledOnceWithExactly(
+        'users.login.failure',
+        { 'usr.id': ' ' },
+        'passportTrackEvent',
+        undefined,
+        'safe'
+      )
+    })
+
+    it('should report login success when passportUser is present', () => {
+      passportModule.passportTrackEvent(loginLocal, userUuid, rootSpan, 'safe')
+
+      expect(setUser.setUserTags).to.have.been.calledOnceWithExactly({ id: userUuid.id }, rootSpan)
+      expect(events.trackEvent).to.have.been.calledOnceWithExactly(
+        'users.login.success',
+        {},
+        'passportTrackEvent',
+        rootSpan,
+        'safe'
+      )
+    })
+
+    it('should report login success and blank id in safe mode when id is not a uuid', () => {
+      const user = userUuid
+
+      user.id = 'publicName'
+      passportModule.passportTrackEvent(loginLocal, user, rootSpan, 'safe')
+
+      expect(setUser.setUserTags).to.have.been.calledOnceWithExactly({ id: ' ' }, rootSpan)
+      expect(events.trackEvent).to.have.been.calledOnceWithExactly(
+        'users.login.success',
+        {},
+        'passportTrackEvent',
+        rootSpan,
+        'safe'
+      )
+    })
+
+    it('should report login success and the extended fields in extended mode', () => {
+      const user = userUuid
+
+      user.id = 'publicName'
```

Comment:
You're modifying the original object here, the first line copies the pointer, and the second line update the field of a pointer, so you're mutating userUuid too and I don't think that's what you're trying to do